slides/workshop_rap_for_managers_basics.qmd

```diff
@@ -0,0 +1,343 @@
+:::
+
+# I'm sure there'll be comments in the code that tell me what to do...
+
+::: {.column width="100%"}
+:::
+
+::: {.column width="100%"}
+![](images/code_nightmare_3.png){fig-align="center" width="50%"}
+:::
+
+# Now there's a weird error
+
+::: {.column width="100%"}
+Time to ask for help!
+:::
+
+::: {.column width="100%"}
+:::
+
+::: {.column width="100%"}
+![](images/code_nightmare_4.jpg){fig-align="center" width="40%"}
+:::
+
+# Of course...
+
+-   This scenario is exaggerated, but it's not far from the truth of what we see in many teams
+-   It's not just about the time it takes to find the code and the stress it might cause, but the risk of errors and the lack of transparency
+-   Imagine how it would feel to encounter this issue as a new starter in the team
+
+# How could RAP have avoided these issues?
+
+# Finding the correct version of the code
+
+-   Sensible folder and file structure <span style="color:#4472c4;">(**Good practice**)</span>
+-   Version controlled final code scripts <span style="color:#70ad47;">(**Great practice**)</span>
+-   Collaboratively develop code using Git <span style="color:#f67400;">(**Best practice**)</span>
+
+Using Azure DevOps to store and version control your code scripts allows you to:
+
+-   Immediately see the current version of any code your team has been using
+-   View all historical versions of the code, along with when the changes were made, what they were, and who made them
+-   Roll back to previous versions of the code if you encounter an issue with the current version
+
+# Understanding how to run the code
+
+-   Documentation <span style="color:#4472c4;">(**Good practice**)</span>
+
+In your Azure DevOps repo, you should have a README file that explains how to run the code, what it does, and any other important information about the code. This means that anyone in your team (including you!) can pick up the code and run it without needing to ask for help.
+
+Your code should also be well-commented, so that anyone reading it can understand what each part of the code does. If sections of code need updating each time, these should be clearly indicated in the code comments.
+
+# Running the code
+
+-   All source data stored in a database <span style="color:#4472c4;">(**Good practice**)</span>
+-   Recyclable code for future use <span style="color:#70ad47;">(**Great practice**)</span>
+
+If your code is set up to read data from a database, you can be sure that the data you're using is up-to-date and accurate. This also means that you can easily update the data you're using without needing to change the code.
+
+Recyclable code is code that can be used in multiple projects. This means that you can write code once and use it in multiple projects, saving time and reducing the risk of errors. This is particularly useful for code that is used in multiple publications, or for code that is used in multiple parts of the same publication.
+
+# What we expect from managers
```

Comment:
it's very RAP focussed which is helpful but I don't think we want to lose the opportunity to push the technical leadership angle too (thinking back to past discussions about if 'RAP' for managers was the right name for this). From my perspective we want them not to be scared of code / put efforts to building their own technical skills (for effective leadership and cover) and building that capability in their teams so they aren't left behind. That includes core skills R, SQL, git (databricks too I guess) + the RAP stuff as a framework for all this in the world of stats / analysis.